micro-rdk-installer/src/nvs/request.rs

```diff
@@ -1,37 +1,30 @@
 use super::super::error::Error;
-use super::data::ViamFlashStorageData;
-
-use rcgen::{date_time_ymd, CertificateParams, DistinguishedName};
 use std::fs::File;
 use std::io::Write;
 use std::path::{Path, PathBuf};
 
-use local_ip_address::local_ip;
-use tokio::runtime::Runtime;
-use viam::gen::proto::app::v1::{
-    robot_service_client::RobotServiceClient, AgentInfo, CertificateRequest, ConfigRequest,
-};
-use viam_rust_utils::rpc::dial::{DialOptions, RPCCredentials};
-
 /*
 This module contains the logic for acquiring the security credentials for a robot from
 the Viam App and preparing for flash storage
 */
 
-const RELEASES_BASE_URL: &str = "https://github.com/viamrobotics/micro-rdk/releases/";
+const RELEASES_BASE_URL: &str = "https://github.com/viamrobotics/micro-rdk/releases";
+const BINARY_NAME: &str = "micro-rdk-server-esp32.bin";
 
 pub async fn download_micro_rdk_release(
     path: &Path,
     version: Option<String>,
 ) -> Result<PathBuf, Error> {
-    let mut release_url: String = RELEASES_BASE_URL.to_owned();
-    match version {
-        Some(version) => {
-            let rest_of_url = format!("/download/{version}/micro-rdk-server-esp32.bin");
-            release_url.push_str(&rest_of_url)
-        }
-        None => release_url.push_str("latest/download/micro-rdk-server-esp32.bin"),
-    };
+    let release_url = version.map_or_else(
+        || {
+            format!(
+                "{}/{}/{}",
+                RELEASES_BASE_URL, "latest/download/", BINARY_NAME
```

Comment:
looks like this will result in two slashes after the word "download"? Or am I missing something?

Reply:
yeah the previous piece of code had it although it was before the download know that I re read it (L30) should I put it there? I don't think the // matter but if it's not needed then we should probably remove it.